src/main/java/com/tribesbackend/tribes/controllers/usercontrollers/UserRestController.java

```diff
@@ -0,0 +1,113 @@
+        System.out.println(newKingdom.getId());
+        kingdomRepo.save(newKingdom);
+        System.out.println(newUser.getId());
+        System.out.println(newUser.getUsername());
+        System.out.println(newKingdom.getId());
+        return new ResponseEntity(new RegistrationResponseJson(newUser.getId(), newUser.getUsername(),
+                newKingdom.getId(), "No avatar yet", 0), HttpStatus.OK);
+
+    }
+
+    @PostMapping(value = "/login")
+    public ResponseEntity loginUser(@RequestBody TribesUser tribesUser) {
+        if (tribesUser.getUsername() == null || tribesUser.getUsername().isEmpty() ||
+                tribesUser.getPassword() == null || tribesUser.getPassword().isEmpty()) {
+            return new ResponseEntity(ErrorMessagesMethods.jsonFieldIsEmpty(tribesUser), HttpStatus.BAD_REQUEST);
+
+
+        } else if (!userTRepository.findTribesUserByUsername(tribesUser.getUsername()).isPresent()) {
+            return new ResponseEntity(ErrorMessagesMethods.notSuchUser(tribesUser.getUsername()), HttpStatus.UNAUTHORIZED);
+        } else if (userTRepository.findTribesUserByUsername(tribesUser.getUsername()).get().getPassword().equals(tribesUser.getPassword())) {
+            userTRepository.findTribesUserByUsername(tribesUser.getUsername()).get().setLoggedIn(true);
+            userTRepository.save(userTRepository.findTribesUserByUsername(tribesUser.getUsername()).get());
+            return new ResponseEntity(
+                    new OKstatus(
+                            JWT.create()
+                                    .withSubject(tribesUser.getUsername())
+                                    .withExpiresAt(new Date(System.currentTimeMillis() + EXPIRATION_TIME))
+                                    .sign(HMAC512(SecurityConstants.SECRET.getBytes())))
+                    , HttpStatus.OK);
+        } else if (!userTRepository.findTribesUserByUsername(tribesUser.getUsername()).get().getPassword()
+                .equals(tribesUser.getPassword())) {
+            return new ResponseEntity(ErrorMessagesMethods.wrongPassword()
+                    , HttpStatus.UNAUTHORIZED);
+        }
+
+        return new ResponseEntity(HttpStatus.CONFLICT);
+    }
+
+    @DeleteMapping(value = "/logout")
+    public ResponseEntity logoutUser() {
+        if (SecurityContextHolder.getContext().getAuthentication().isAuthenticated()) {
+            userTRepository.findTribesUserByUsername(SecurityContextHolder.getContext().getAuthentication().getName())
+                    //uncatched get
+                    .get().setLoggedIn(false);
+            return new ResponseEntity(new LogoutMessages("Logged out successfully!"), HttpStatus.OK);
+        } else
+            return new ResponseEntity(new LogoutMessages("Unauthorized request!"), HttpStatus.FORBIDDEN);
+    }
+
+    @GetMapping(value = "/user/testjwt")
+    public String testingEndpoint() {
+        return SecurityContextHolder.getContext().getAuthentication().getName();
```

Comment:
@Kopsova we should have a `BaseController` that has autowired kindomrepo/userrepo and method `getCurrentKingdom()`

Reply:
Done